### PR TITLE
Added proptests for VariableIter

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
- Added proptests for `VariableIter`
- Fixed a bug found by proptests

### CODE REVIEW

File: mqtt/mqtt-broker/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,6 @@ humantime-serde = "1.0"
 lazy_static = "1.4"
 openssl = "0.10"
 pin-project = "0.4"
-proptest = { version = "0.9", optional = true }
```
Comment (dmolokanov) [inline mqtt/mqtt-broker/Cargo.toml:21]: I guess without this line we cannot run model based tests. could you check please?

**Review by dmolokanov — APPROVED** [review]
lgtm

File: mqtt/mqtt-policy/Cargo.toml [hunk 2]
```diff
@@ -17,4 +17,5 @@ policy = { path = "../policy" }
 [dev-dependencies]
 assert_matches = "1.3"
 bytes = "0.5"
+proptest = "0.9"
```
Comment (dmolokanov) [inline mqtt/mqtt-policy/Cargo.toml:20]: I guess you can remove this line, because you already have the optional include back?

Comment (vadim-kovalyov) [reply mqtt/mqtt-policy/Cargo.toml:20]: those are different Cargo.toml files. one from broker, another from policy.

**Review by dmolokanov — APPROVED** [review]
More proptests please!

**Pull request merged**